x-pack/plugins/triggers_actions_ui/public/application/sections/actions_connectors_list/components/actions_connectors_home.test.tsx

```diff
@@ -0,0 +1,215 @@
+  });
+
+  it('there are Connectors and Logs tabs', async () => {
+    const props: RouteComponentProps<MatchParams> = {
+      history: createMemoryHistory({
+        initialEntries: ['/connectors'],
+      }),
+      location: createLocation('/connectors'),
+      match: {
+        isExact: true,
+        path: '/connectors',
+        url: '',
+        params: {
+          section: 'connectors',
+        },
+      },
+    };
+
+    render(
+      <IntlProvider locale="en">
+        <Router history={props.history}>
+          <QueryClientProvider client={queryClient}>
+            <ActionsConnectorsHome {...props} />
+          </QueryClientProvider>
+        </Router>
+      </IntlProvider>
+    );
+
+    const tabs = await screen.findAllByRole('tab');
+    expect(tabs).toHaveLength(2);
+    expect(tabs[0]).toHaveTextContent('Connectors');
+    expect(tabs[1]).toHaveTextContent('Logs');
+  });
+
+  it('show "Create connector" and "Documentation" buttons when on Connectors tab', async () => {
+    const props: RouteComponentProps<MatchParams> = {
+      history: createMemoryHistory({
+        initialEntries: ['/connectors'],
+      }),
+      location: createLocation('/connectors'),
+      match: {
+        isExact: true,
+        path: '/connectors',
+        url: '',
+        params: {
+          section: 'connectors',
+        },
+      },
+    };
+
+    render(
+      <IntlProvider locale="en">
+        <Router history={props.history}>
+          <QueryClientProvider client={queryClient}>
+            <ActionsConnectorsHome {...props} />
+          </QueryClientProvider>
+        </Router>
+      </IntlProvider>
+    );
+
+    const createConnectorButton = await screen.findByRole('button', { name: 'Create connector' });
+    expect(createConnectorButton).toBeEnabled();
+
+    const documentationButton = await screen.findByRole('link', { name: 'Documentation' });
+    expect(documentationButton).toBeEnabled();
+  });
+
+  it('hide "Create connector" button when on Logs tab', async () => {
+    const props: RouteComponentProps<MatchParams> = {
+      history: createMemoryHistory({
+        initialEntries: ['/logs'],
+      }),
+      location: createLocation('/logs'),
+      match: {
+        isExact: true,
+        path: `/logs`,
+        url: '',
+        params: {
+          section: 'logs',
+        },
+      },
+    };
+
+    render(
+      <IntlProvider locale="en">
+        <Router history={props.history}>
+          <QueryClientProvider client={queryClient}>
+            <ActionsConnectorsHome {...props} />
+          </QueryClientProvider>
+        </Router>
+      </IntlProvider>
+    );
+
+    const documentationButton = await screen.findByRole('link', { name: 'Documentation' });
+    expect(documentationButton).toBeEnabled();
+
+    await waitFor(() => {
+      expect(screen.queryByRole('button', { name: 'Create connector' })).not.toBeInTheDocument();
+    });
```

Comment:
```suggestion
    expect(screen.queryByRole('button', { name: 'Create connector' })).not.toBeInTheDocument();
```

I don't think you need the `waitFor` in this case.

Reply:
@adcoelho I had doubts here. I render the component. So should wait until it will be render. Everywhere I used find in the same situation. But here I cannot because if queryBy, so I used waitFor to follow the same logic.
So you do not think I need for render to happened? How does it work?

Reply:
> So should wait until it will be render. Everywhere I used find in the same situation. But here I cannot because if queryBy, so I used waitFor to follow the same logic.
So you do not think I need for render to happened? How does it work?

I am not sure I understand what you are trying to say 😅 but all `waitFor` does is run the callback you pass it until it either does not `throw` or a timeout happens.

In your test, the condition `expect(screen.queryByRole('button', ...)).not.toBeInTheDocument();` will never throw because the button is not there so it is pointless to have the `waitFor`.

As a side note, I don't think "waiting for render" as you said applies. [The documentation](https://testing-library.com/docs/react-testing-library/api#render) as an example where `getBy` is used and no `waitFor` is there to "wait for render".

```
test('renders a message', () => {
  const {asFragment, getByText} = render(<Greeting />)
  expect(getByText('Hello, world!')).toBeInTheDocument()
```

It is useful if you have lots of async stuff going on that you need to wait for before something happens but it doesn't mean nothing rendered.